bag_unpacker/src/main/scala/uk/ac/wellcome/platform/archive/bagunpacker/services/Unpacker.scala

```diff
@@ -6,21 +6,26 @@ import java.time.Instant
 
 import com.amazonaws.services.s3.AmazonS3
 import org.apache.commons.compress.archivers.ArchiveEntry
-import uk.ac.wellcome.platform.archive.bagunpacker.models.UnpackSummary
-import uk.ac.wellcome.platform.archive.bagunpacker.storage.Archive
+import uk.ac.wellcome.platform.archive.bagunpacker.config.models.UnpackerConfig
 import uk.ac.wellcome.platform.archive.common.operation.services.{
   OperationFailure,
   OperationResult
 }
-import uk.ac.wellcome.storage.ObjectLocation
+import uk.ac.wellcome.platform.archive.bagunpacker.models.UnpackSummary
+import uk.ac.wellcome.platform.archive.bagunpacker.storage.Archive
 import uk.ac.wellcome.platform.archive.common.ConvertibleToInputStream._
+import uk.ac.wellcome.storage.ObjectLocation
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
-class Unpacker(implicit s3Client: AmazonS3, ec: ExecutionContext) {
+class Unpacker(config: UnpackerConfig)(implicit s3Client: AmazonS3,
+                                       ec: ExecutionContext) {
 
-  private val s3Uploader = new S3Uploader()
+  // See comments inside both of these classes -- it's important that they
+  // both have the same buffer size.
+  private val s3Uploader = new S3Uploader(bufferSize = config.bufferSize)
```

Comment:
from the SDK best practice docs should this be `bufferSize+1`?

Reply:
Sure